crates/holochain_zome_types/src/zome.rs

```diff
@@ -132,6 +133,20 @@ impl From<CoordinatorZome> for CoordinatorZomeDef {
     }
 }
 
+/// A zome defined by a preserialized
+/// wasmer module, as a "dynamic library" (dylib)
+/// Useful for iOS
+#[derive(Serialize, Deserialize, Hash, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct WasmZomeDylib {
+    /// The WasmHash representing the WASM byte code for this zome.
+    pub wasm_hash: holo_hash::WasmHash,
+    /// The path to the dylib
+    pub path: PathBuf,
+    /// The zome dependencies
+    pub dependencies: Vec<ZomeName>,
+}
```

Comment:
@Connoropolous since the only difference between this and the existing `ZomeDef` is the path, what if you just add a field to `WasmZome` thus:

```rust
#[serde(default)]
dylib_path: Option<PathBuf>,
```

If it works, this would make the change non-breaking. If it's a breaking change, it breaks not only the database, but forced a recompile of every existing DNA, invalidating all existing source chains.

Reply:
(the existence of `dylib_path` would indicate the difference in zome type)

Reply:
Unless you're specifically relying on the enum for serialization or something, collapsing the two enum variants and adding the Option preserves the information exactly, so I feel confident that it will work out.

Reply:
@Connoropolous I'd still prefer this change but I just heard that this is being scheduled for 0.2.0, and if so the concerns about the breaking change are not so large given the version bump. So that slightly shifts the priorities.

Want me to do this refactor today?

Reply:
Yes, amazing!

Reply:
@Connoropolous I made a PR to your branch, you can merge that into yours if it works for you. I'm getting seemingly unrelated errors, maybe I need to do a git clean, but hoping you can verify that it works for you. https://github.com/Sprillow/holochain/pull/2

Also here is the branch against develop which could be helpful since it undoes a lot of changes you did in your branch around the new variant https://github.com/holochain/holochain/pull/2216

Reply:
I will also pull that cranelift fn into holochain_types, though unfortunately the errors I'm finding are missing holochain_wasmer imports so it might hard to tell if I did it properly